app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,30 @@
+class PostsController < ApplicationController
+  def new
+    @post = current_user.posts.build
```

Comment:
better to say '@post = Post.new`

Reply:
You're doing the current_user.posts.build in create, this is not necessary. `@post = Post.new` is simply to initialise the @post variable type